src/user/wikis/Editor/Editor.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from "react";
+import DeleteOutlinedIcon from "@material-ui/icons/DeleteOutlined";
+import CancelButton from "@material-ui/icons/ClearOutlined";
+import SaveButton from "@material-ui/icons/SaveOutlined";
+import "react-mde/lib/styles/css/react-mde-all.css";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import * as Showdown from "showdown";
+import ReactMde from "react-mde";
+import "./Editor.scss";
+
+class Editor extends Component {
+  state = {
+    comments: "",
+    selectedTab: "write",
+    title: this.props.page?.title,
+    content: this.props.page?.content,
+  };
+
+  handleSave = () => {
+    this.props.save(
+      {
+        title: this.state.title,
+        content: this.state.content,
+        comments: this.state.comments,
+        history: this.props.page?.history
+      },
+      this.props.newPage,
+      this.props.sidebar
+    );
+  };
+  
+  setContent = content => this.setState({ content });
+  
+  setTitle = evt => this.setState({ title: evt.target.value });
+
+  setSelectedTab = selectedTab => this.setState({ selectedTab });
+
+  setComments = evt => this.setState({ comments: evt.target.value });
+
+  render() {
+    const converter = new Showdown.Converter({
+      tables: true,
+      tasklists: true,
+      strikethrough: true,
+      simplifiedAutoLink: true,
+    });
+
+    return (
+      <div className="wiki-editor">
+        <div className="wikis-top-controls">
+          <Button 
+            variant="outline-danger" 
+            onClick={this.props.delete} 
```

Comment:
destructure these property for better flexibility

Reply:
Applicable for whole codebase

Reply:
Done